internal/provider/edgegw/nat_rule_types.go

```diff
@@ -0,0 +1,115 @@
+			Enabled:                supertypes.NewBoolNull(),
+			ExternalAddress:        supertypes.NewStringNull(),
+			FirewallMatch:          supertypes.NewStringNull(),
+			ID:                     supertypes.NewStringUnknown(),
+			InternalAddress:        supertypes.NewStringNull(),
+			Name:                   supertypes.NewStringUnknown(),
+			Priority:               supertypes.NewInt64Null(),
+			RuleType:               supertypes.NewStringNull(),
+			SnatDestinationAddress: supertypes.NewStringNull(),
+		}
+
+	case tfsdk.Plan:
+		return &NATRuleModel{
+			Description:            supertypes.NewStringNull(),
+			DnatExternalPort:       supertypes.NewStringNull(),
+			EdgeGatewayID:          supertypes.NewStringUnknown(),
+			EdgeGatewayName:        supertypes.NewStringUnknown(),
+			Enabled:                supertypes.NewBoolNull(),
+			ExternalAddress:        supertypes.NewStringNull(),
+			FirewallMatch:          supertypes.NewStringNull(),
+			ID:                     supertypes.NewStringUnknown(),
+			InternalAddress:        supertypes.NewStringNull(),
+			Name:                   supertypes.NewStringUnknown(),
+			Priority:               supertypes.NewInt64Null(),
+			RuleType:               supertypes.NewStringNull(),
+			SnatDestinationAddress: supertypes.NewStringNull(),
+		}
+
+	case tfsdk.Config:
+		return &NATRuleModel{
+			Description:            supertypes.NewStringNull(),
+			DnatExternalPort:       supertypes.NewStringNull(),
+			EdgeGatewayID:          supertypes.NewStringUnknown(),
+			EdgeGatewayName:        supertypes.NewStringUnknown(),
+			Enabled:                supertypes.NewBoolNull(),
+			ExternalAddress:        supertypes.NewStringNull(),
+			FirewallMatch:          supertypes.NewStringNull(),
+			ID:                     supertypes.NewStringUnknown(),
+			InternalAddress:        supertypes.NewStringNull(),
+			Name:                   supertypes.NewStringUnknown(),
+			Priority:               supertypes.NewInt64Null(),
+			RuleType:               supertypes.NewStringNull(),
+			SnatDestinationAddress: supertypes.NewStringNull(),
+		}
+
+	default:
+		panic(fmt.Sprintf("unexpected type %T %v", t, x))
+	}
+}
+
+func (rm *NATRuleModel) Copy() *NATRuleModel {
+	x := &NATRuleModel{}
+	utils.ModelCopy(rm, x)
+	return x
+}
+
+func (rm *NATRuleModel) ToNsxtNatRule(ctx context.Context) (values *govcdtypes.NsxtNatRule, err error) {
+	values = &govcdtypes.NsxtNatRule{
+		Name:                     rm.Name.Get(),
+		Description:              rm.Description.Get(),
+		Enabled:                  rm.Enabled.Get(),
+		ExternalAddresses:        rm.ExternalAddress.Get(),
+		InternalAddresses:        rm.InternalAddress.Get(),
+		SnatDestinationAddresses: rm.SnatDestinationAddress.Get(),
+		DnatExternalPort:         rm.DnatExternalPort.Get(),
+		Type:                     rm.RuleType.Get(),
+		FirewallMatch:            rm.FirewallMatch.Get(),
+		Priority:                 utils.TakeIntPointer(int(rm.Priority.Get())),
```

Comment:
In the attribute `Priority` the int pointer is used for differentiate `nil` and `0` value

Actually if Priority is not set `0` value is set and not `nil`

Reply:
This value is compute, and the default value return by api is value 0.

Reply:
Now the value set in plan/state is unknow: Priority:               supertypes.NewInt64Unknown()